include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp

```diff
@@ -405,13 +409,13 @@ namespace picongpu
                                 j = 1;
                             }
 
-                            float3_X iIdxForw = float3_X(totalCellIdx[0], totalCellIdx[1], totalCellIdx[2]);
+                            float3_X iIdxForw = makeVector3(totalCellIdx);
                             iIdxForw[i] = totalCellIdx[i] + 0.5_X;
-                            float3_X iIdxBackw = float3_X(totalCellIdx[0], totalCellIdx[1], totalCellIdx[2]);
+                            float3_X iIdxBackw = makeVector3(totalCellIdx);
                             iIdxBackw[i] = totalCellIdx[i] - 0.5_X;
-                            float3_X jIdxForw = float3_X(totalCellIdx[0], totalCellIdx[1], totalCellIdx[2]);
+                            float3_X jIdxForw = makeVector3(totalCellIdx);
                             jIdxForw[j] = totalCellIdx[j] + 0.5_X;
-                            float3_X jIdxBackw = float3_X(totalCellIdx[0], totalCellIdx[1], totalCellIdx[2]);
+                            float3_X jIdxBackw = makeVector3(totalCellIdx);
```

Comment:
Well this might be one of the problems this implementation has and why it did not work. Actually, I think the best solution would be to just delete `struct DispersivePulseFunctorIncidentB` completely.
It is not used anyway.
That is, there is no fix required, but a cleanup :sweat_smile:

Reply:
@steindev What should I make out of your comments? They are not rejecting, but sound like you actually would like to requite changes. This pull request could be merged now if you do not object.

Reply:
I opened #4932 where the B-field implementation is deleted. Please merge this or the other PR.